test/membrane_http_adaptive_stream/integration_test/sink_bin_integration_test.exs

```diff
@@ -0,0 +1,159 @@
+
+    @impl true
+    def handle_init(options) do
+      sink_bin = %SinkBin{
+        muxer_segment_duration: 2 |> Membrane.Time.seconds(),
+        manifest_module: HLS,
+        target_window_duration: 30 |> Membrane.Time.seconds(),
+        target_segment_duration: 2 |> Membrane.Time.seconds(),
+        persist?: false,
+        storage: %FileStorage{
+          directory: options[:output_dir]
+        }
+      }
+
+      elements =
+        options[:sources]
+        |> Enum.map(fn {source, encoding, track_name} ->
+          [
+            {{:source, track_name}, source},
+            {{:parser, encoding, track_name}, get_parser_for_encoding(encoding)}
+          ]
+        end)
+
+      links =
+        elements
+        |> Enum.map(fn [
+                         {{:source, name} = source_id, _},
+                         {{:parser, encoding, name} = parser_id, _}
+                       ] ->
+          link(source_id)
+          |> to(parser_id)
+          |> via_in(Pad.ref(:input, name), options: [encoding: encoding, track_name: name])
+          |> to(:sink_bin)
+        end)
+
+      children = elements |> List.flatten() |> Enum.into(%{:sink_bin => sink_bin})
+
+      {{:ok, spec: %ParentSpec{children: children, links: links}}, %{}}
+    end
+
+    defp get_parser_for_encoding(encoding) do
+      case encoding do
+        :H264 ->
+          %H264_Parser{
+            framerate: {30, 1},
+            alignment: :au,
+            attach_nalus?: true
+          }
+
+        :AAC ->
+          %AAC_Parser{
+            out_encapsulation: :none
+          }
+      end
+    end
+  end
+
+  test "single video track" do
+    test_pipeline(
+      @single_video_track_source,
+      @single_video_track_ref_path,
+      @single_video_track_test_path
+    )
+  end
+
+  test "audio video tracks" do
+    test_pipeline(
+      @audio_video_tracks_sources,
+      @audio_video_tracks_ref_path,
+      @audio_video_tracks_test_path
+    )
+  end
+
+  defp run_pipeline(sources, result_directory) do
+    {:ok, pipeline} =
+      %Testing.Pipeline.Options{
+        module: TestPipeline,
+        custom_args: %{
+          sources: sources,
+          output_dir: result_directory
+        }
+      }
+      |> Testing.Pipeline.start_link()
+
+    Testing.Pipeline.play(pipeline)
+    assert_pipeline_playback_changed(pipeline, _, :playing)
+
+    # For every input track pipeline receives two :end_of_stream notifications,
+    # one from :sink_bin, and the other from :sink - :sink_bin inner child
+    for _source <- sources,
+        _end_of_stream_notification_count <- 1..2,
+        do: assert_end_of_stream(pipeline, :sink_bin)
+
+    Testing.Pipeline.stop_and_terminate(pipeline, blocking?: true)
+    assert_pipeline_playback_changed(pipeline, _, :stopped)
+  end
+
+  defp test_pipeline(sources, reference_directory, test_directory) do
+    if @create_references_mode do
+      run_pipeline(sources, reference_directory)
+    end
```

Comment:
I strongly object. `@create_references_mode` is effectively a switch for turning tests into `assert A == A` and it is way too easy to forget to disable this mode. I would rather extract pipeline module to another file and just run the pipeline manually from `iex` whenever you need to generate the fixtures. Or run the tests as they are and drag&drop the artefacts

Reply:
We discussed it with @andpodob and came to a conclusion that a test that fails if this mode is enabled is necessary